src/rpc/core_rpc_server.cpp

```diff
@@ -2370,13 +2645,231 @@ namespace cryptonote
       error_resp.message = "Failed to prune blockchain";
       return false;
     }
+    res.status = CORE_RPC_STATUS_OK;
+    return true;
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_rpc_access_info(const COMMAND_RPC_ACCESS_INFO::request& req, COMMAND_RPC_ACCESS_INFO::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
+  {
+    RPC_TRACKER(rpc_access_info);
+
+    bool r;
+    if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_ACCESS_INFO>(invoke_http_mode::JON, "rpc_access_info", req, res, r))
+      return r;
+
+    // if RPC payment is not enabled
+    if (m_rpc_payment == NULL)
+    {
+      res.diff = 0;
+      res.credits_per_hash_found = 0;
+      res.credits = 0;
+      res.height = 0;
+      res.status = CORE_RPC_STATUS_OK;
+      return true;
+    }
+
+    crypto::public_key client;
+    if (!epee::string_tools::hex_to_pod(req.client.substr(0, 2 * sizeof(client)), client))
```

Comment:
The client auth wasn't validated ...

Reply:
IIRC this could was originally called without a signature, just a "hey, new client, here's my pubkey". I think it now has a sig too, I'll have to check. I don't think it matters here since it's not a "do something for me" call, but a "let me know your payment config" call.

Reply:
IIRC this could was originally called without a signature, just a "hey, new client, here's my pubkey". I think it now has a sig too, I'll have to check. I don't think it matters here since it's not a "do something for me" call, but a "let me know your payment config" call.